ast_tools/passes/base.py

```diff
@@ -24,12 +21,8 @@ def __call__(self, args: PASS_ARGS_T) -> PASS_ARGS_T:
 
     @abstractmethod
     def rewrite(self,
-                tree: ast.AST,
+                tree: cst.CSTNode,
                 env: SymbolTable,
                 metadata: tp.MutableMapping,
                 ) -> PASS_ARGS_T:
-        """
-        Type annotation here should be followed except on terminal passes e.g.
-        end_rewite
-        """
-        pass
+        return tree, env, metadata
```

Comment:
Should this be pass instead of return since it's an abstractmethod (i.e. it must be defined in the subclass.  Perhaps having the return is useful for type checking and tool integration?

Reply:
Type checking was my motivation.  I am planning on standing up a typing environment when I get some free cycles.

Reply:
It's really amazing how many stupid bugs I catch with type checking when working on libcst.